arangod/MMFiles/MMFilesCollection.cpp

```diff
@@ -3053,6 +3056,149 @@ void MMFilesCollection::removeLocalDocumentId(LocalDocumentId const& documentId,
   }
 }
 
+namespace {
+Result countDocumentsIterator(MMFilesMarker const* marker,
+                              void* counter, MMFilesDatafile*) {
+  Result res;
```

Comment:
This looks overly expensive for increasing a simple counter.
Given there are 10M documents, this will create 10M unused Result objects. Can we avoid this overhead?